src/main/java/frc/robot/subsystems/HandlerSubsystem.java

```diff
@@ -0,0 +1,122 @@
+package frc.robot.subsystems;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+import com.revrobotics.REVLibError;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.spark.SparkBase.ResetMode;
+import com.revrobotics.spark.SparkLowLevel.MotorType;
+import com.revrobotics.spark.SparkMax;
+import com.revrobotics.spark.config.ClosedLoopConfig.FeedbackSensor;
+import com.revrobotics.spark.config.SparkBaseConfig.IdleMode;
+import com.revrobotics.spark.config.SparkMaxConfig;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Constants.HandlerConstants;
+
+public class HandlerSubsystem extends SubsystemBase {
+  private enum State {
+    EMPTY,
+    INTAKING,
+    LOADING,
+    LOADED,
+    EJECTING,
+    CANCELLING
+  }
+
+  private State m_state;
+  private final SparkMax m_motor;
+  private final RelativeEncoder m_encoder;
+
+  private double m_intakeSpeed = HandlerConstants.kIntakeSpeed;
+  private double m_ejectSpeed = HandlerConstants.kEjectSpeed;
```

Comment:
We should either remove these variables (use the constants directly), or make them `final`.